ddev/tests/cli/meta/scripts/test_upgrade_python.py

```diff
@@ -0,0 +1,43 @@
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from collections import defaultdict
+
+from ddev.repo.constants import PYTHON_VERSION
+
+
+def test_upgrade_python(ddev, repository):
+    major, minor = PYTHON_VERSION.split('.')
+    new_version = f'{major}.{int(minor) + 1}'
+
+    changes = defaultdict(list)
+    for entry in repository.path.iterdir():
+        config_file = entry / 'hatch.toml'
+        if not config_file.is_file():
+            continue
+
+        for i, line in enumerate(config_file.read_text().splitlines()):
+            if line.startswith('python = [') and PYTHON_VERSION in line:
+                changes[config_file].append(i)
+
+    minimum_changes = sum(map(len, changes.values()))
+
+    # TODO: uncomment after merging
```

Comment:
This is due to the way the fixture works, it gives us a clean repository based on upstream